ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -234,6 +235,7 @@ protected ObjectMapper getJsonMapper() {
     // superclass creates a new json mapper on every call
     // we should probably ony create one per application
     ObjectMapper jsonMapper = super.getJsonMapper();
+    jsonMapper.registerModule(new StructSerializationModule());
```

Comment:
thread-satefy:

Maybe, rather than having this `registerModule` call throughput the code, what we could do is have one single place where we initialize the `ObjectMapper` with the configuration we need, and use this everywhere else. This way, we only have one `ObjectMapper` instance, (which is thread-safe once initialised), and any other configuration changes we need to make, (such as plugging in other serializers, can be done in one place).

This also reduces the chance of threading issues due to mappers being modified while concurrently being used.

Reply:
Moved ObjectMapper as a class variable and reused it everywhere.